### PR TITLE
Custom Guide Shape

### PR DESCRIPTION
**Pull request by NicholasMata**
Extends existing functionality to allow for a custom guide shape. 

This is done through the use of a new interface called `Targetable`.

### How it works
If the target view given to the `GuideView` constructor implements `Targetable`. 
* `boundingRect` will be used to set `targetRect` property inside the `GuideView` class. 
* `guidePath` will be used when `onDraw` is called in `GuideView` instead of always using `drawRoundRect` the canvas will call `drawPath` and will draw the guide path.

**PS: I have added onto the sample application with a demonstration of how this works as well.**

This implementation could be used to resolve issue #29

### CODE REVIEW

**Comment by mreram** [comment]
Thanks for your contribution, I will review this ASAP.

File: app/src/main/java/ir/smartdevelop/eram/showcaseview/CircleView.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import android.graphics.Paint;
+import android.graphics.Path;
+import android.graphics.RectF;
+import android.util.AttributeSet;
+import android.view.View;
+
+import smartdevelop.ir.eram.showcaseviewlib.Targetable;
+
+public class CircleView extends View implements Targetable {
+    private static final int DEFAULT_CIRCLE_COLOR = Color.RED;
```
Comment (mreram) [inline app/src/main/java/ir/smartdevelop/eram/showcaseview/CircleView.java:15]: Add a space above this

File: app/src/main/java/ir/smartdevelop/eram/showcaseview/CircleView.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
```
Comment (mreram) [inline app/src/main/java/ir/smartdevelop/eram/showcaseview/CircleView.java:16]: Please remove this space

File: app/src/main/java/ir/smartdevelop/eram/showcaseview/CircleView.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    private int circleColor = DEFAULT_CIRCLE_COLOR;
```
Comment (mreram) [inline app/src/main/java/ir/smartdevelop/eram/showcaseview/CircleView.java:17]: Please add an underline before `circleColor` like other fields or remove other underlines.

**Review by mreram — CHANGES_REQUESTED** [review]
Please review my comments.

File: app/src/main/java/ir/smartdevelop/eram/showcaseview/CircleView.java (hunk 3 above)
Comment (NicholasMata) [reply app/src/main/java/ir/smartdevelop/eram/showcaseview/CircleView.java]: Nice catch

**Comment by NicholasMata** [comment]
If you could review again I made the requested changes. Let me know if I need to do anything else.

**Pull request merged**